fastestimator/op/numpyop/univariate/onehot.py

```diff
@@ -55,11 +55,15 @@ def forward(self, data: List[Union[int, np.ndarray]], state: Dict[str, Any]) ->
         return [self._apply_onehot(elem) for elem in data]
 
     def _apply_onehot(self, data: Union[int, np.ndarray]) -> np.ndarray:
-        class_index = np.array(data)
-        assert "int" in str(class_index.dtype)
-        assert class_index.size == 1, "data must have only one item"
-        class_index = class_index.item()
-        assert class_index < self.num_classes, "label value should be smaller than num_classes"
-        output = np.full((self.num_classes), fill_value=self.label_smoothing / self.num_classes, dtype="float32")
-        output[class_index] = 1.0 - self.label_smoothing + self.label_smoothing / self.num_classes
+        data = np.array(data)
+        assert "int" in str(data.dtype)
```

Comment:
add an assert message saying the data type must be an integer

Reply:
added assert message